### PR TITLE
feature/jl/thursday_hardening

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

File: sonar-ws/src/main/java/org/sonarqube/ws/client/user/UsersWsParameters.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+public class UsersWsParameters {
+
+  public static final String CONTROLLER_USERS = "api/users";
+
+  public static final String ACTION_CREATE = "create";
+  public static final String ACTION_UPDATE = "update";
+
+  public static final String PARAM_LOGIN = "login";
+  public static final String PARAM_PASSWORD = "password";
```
Comment (SonarTech) [inline sonar-ws/src/main/java/org/sonarqube/ws/client/user/UsersWsParameters.java]: BLOCKER Remove this hard-coded password. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2068)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* BLOCKER 1 blocker
* MAJOR 2 major

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [WsUtils.java#L43](https://github.com/SonarSource/sonarqube/blob/2ac2ac67bac48702575525535010ce16e6945856/server/sonar-server/src/main/java/org/sonar/server/ws/WsUtils.java#L43): Remove this unused "LOG" private field. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1068)
1. MAJOR [user](https://github.com/SonarSource/sonarqube/blob/2ac2ac67bac48702575525535010ce16e6945856/sonar-ws/src/main/java/org/sonarqube/ws/client/user): Add a 'package-info.java' file to document the 'user' package [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1228)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* BLOCKER 1 blocker
* MAJOR 2 major

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [WsUtils.java#L43](https://github.com/SonarSource/sonarqube/blob/03b2a6e92fd261b7ffb0fd8555cec122cfffb9a1/server/sonar-server/src/main/java/org/sonar/server/ws/WsUtils.java#L43): Remove this unused "LOG" private field. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1068)
1. MAJOR [user](https://github.com/SonarSource/sonarqube/blob/03b2a6e92fd261b7ffb0fd8555cec122cfffb9a1/sonar-ws/src/main/java/org/sonarqube/ws/client/user): Add a 'package-info.java' file to document the 'user' package [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1228)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* BLOCKER 1 blocker
* MAJOR 2 major

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [WsUtils.java#L43](https://github.com/SonarSource/sonarqube/blob/046a186ca065da12c3103dab47d5b871b514ff93/server/sonar-server/src/main/java/org/sonar/server/ws/WsUtils.java#L43): Remove this unused "LOG" private field. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1068)
1. MAJOR [user](https://github.com/SonarSource/sonarqube/blob/046a186ca065da12c3103dab47d5b871b514ff93/sonar-ws/src/main/java/org/sonarqube/ws/client/user): Add a 'package-info.java' file to document the 'user' package [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1228)

File: server/sonar-server/src/main/java/org/sonar/server/user/ws/CreateAction.java [hunk 2]
```diff
@@ -20,6 +20,8 @@
 package org.sonar.server.user.ws;
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/user/ws/CreateAction.java]: you haven't added a test for `CreateActionTest` and `UpdateActionTest` to cover the use case you're solving, which means several scm accounts, with one of the users having a comma in its scm account

**Comment by julienlancelot** [comment]
@teryk Feedbacks applied, please review them !

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* BLOCKER 1 blocker
* MAJOR 2 major

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [WsUtils.java#L43](https://github.com/SonarSource/sonarqube/blob/a53eeb56ef21656f097fd5362f4428e6eefde168/server/sonar-server/src/main/java/org/sonar/server/ws/WsUtils.java#L43): Remove this unused "LOG" private field. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1068)
1. MAJOR [user](https://github.com/SonarSource/sonarqube/blob/a53eeb56ef21656f097fd5362f4428e6eefde168/sonar-ws/src/main/java/org/sonarqube/ws/client/user): Add a 'package-info.java' file to document the 'user' package [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1228)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* BLOCKER 1 blocker
* MAJOR 2 major

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [WsUtils.java#L43](https://github.com/SonarSource/sonarqube/blob/8e3f9410ec541e69a0c2aff14d1adf5ce405ebe0/server/sonar-server/src/main/java/org/sonar/server/ws/WsUtils.java#L43): Remove this unused "LOG" private field. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1068)
1. MAJOR [user](https://github.com/SonarSource/sonarqube/blob/8e3f9410ec541e69a0c2aff14d1adf5ce405ebe0/sonar-ws/src/main/java/org/sonarqube/ws/client/user): Add a 'package-info.java' file to document the 'user' package [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1228)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* BLOCKER 1 blocker
* MAJOR 2 major

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [WsUtils.java#L43](https://github.com/SonarSource/sonarqube/blob/b06dde3fe95d61bd6c893947f7d5fba6263614e2/server/sonar-server/src/main/java/org/sonar/server/ws/WsUtils.java#L43): Remove this unused "LOG" private field. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1068)
1. MAJOR [user](https://github.com/SonarSource/sonarqube/blob/b06dde3fe95d61bd6c893947f7d5fba6263614e2/sonar-ws/src/main/java/org/sonarqube/ws/client/user): Add a 'package-info.java' file to document the 'user' package [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1228)

**Pull request merged**